<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1764</width>
    <height>798</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="OpenGLWidget" name="openGLWidget" native="true">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>50</y>
      <width>768</width>
      <height>581</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>1600</x>
      <y>50</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Ray Trace</string>
    </property>
   </widget>
   <widget class="ImageViewer" name="imageViewer" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>768</width>
      <height>581</height>
     </rect>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>1590</x>
      <y>100</y>
      <width>120</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>Shade Model</string>
    </property>
    <widget class="QRadioButton" name="radioButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>82</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Lambertian</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioButton_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>82</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Blinn-Phong</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>1600</x>
      <y>200</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Save Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>1600</x>
      <y>230</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Load Background</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>1590</x>
      <y>290</y>
      <width>120</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>Materials</string>
    </property>
    <widget class="MaterialComboBox" name="materialComboBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <property name="editable">
      <bool>false</bool>
     </property>
     <property name="currentIndex">
      <number>-1</number>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_4">
    <property name="geometry">
     <rect>
      <x>1600</x>
      <y>390</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Load Obj</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox">
    <property name="geometry">
     <rect>
      <x>1600</x>
      <y>450</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>1600</x>
      <y>430</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Antialias Sample Size</string>
    </property>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>660</y>
      <width>118</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_5">
    <property name="geometry">
     <rect>
      <x>1600</x>
      <y>550</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear Screen</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_2">
    <property name="geometry">
     <rect>
      <x>1600</x>
      <y>500</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="value">
     <number>1</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>1600</x>
      <y>480</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Depth Of Field Samples</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1764</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenGLWidget</class>
   <extends>QWidget</extends>
   <header location="global">openglwidget.h</header>
   <container>1</container>
   <slots>
    <slot>raytraceImage()</slot>
    <slot>changeShadeModelLambertian(bool)</slot>
    <slot>changeShadeModelBlinnPhong(bool)</slot>
    <slot>saveImage()</slot>
    <slot>loadBackgroundTexture()</slot>
    <slot>changeMaterial(int)</slot>
    <slot>loadObjFile()</slot>
    <slot>changeAntiAlias(int)</slot>
    <slot>clearScreen()</slot>
    <slot>changeDepthOfField(int)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>ImageViewer</class>
   <extends>QWidget</extends>
   <header location="global">imageviewer.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>MaterialComboBox</class>
   <extends>QComboBox</extends>
   <header location="global">materialcombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>raytraceImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1644</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>1189</x>
     <y>91</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>openGLWidget</receiver>
   <slot>changeShadeModelLambertian(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1618</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>1225</x>
     <y>162</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_2</sender>
   <signal>toggled(bool)</signal>
   <receiver>openGLWidget</receiver>
   <slot>changeShadeModelBlinnPhong(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1655</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>1217</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>saveImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1641</x>
     <y>255</y>
    </hint>
    <hint type="destinationlabel">
     <x>1187</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>loadBackgroundTexture()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1618</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>1169</x>
     <y>277</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>materialComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>changeMaterial(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1603</x>
     <y>374</y>
    </hint>
    <hint type="destinationlabel">
     <x>1472</x>
     <y>371</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>loadObjFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1624</x>
     <y>427</y>
    </hint>
    <hint type="destinationlabel">
     <x>1484</x>
     <y>441</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>changeAntiAlias(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1623</x>
     <y>504</y>
    </hint>
    <hint type="destinationlabel">
     <x>1476</x>
     <y>490</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>clearScreen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1628</x>
     <y>605</y>
    </hint>
    <hint type="destinationlabel">
     <x>1491</x>
     <y>545</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>changeDepthOfField(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1613</x>
     <y>545</y>
    </hint>
    <hint type="destinationlabel">
     <x>1546</x>
     <y>559</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
